3D Tank/Assets/Scripts/GameController.cs

```diff
@@ -0,0 +1,39 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class GameController<T> : MonoBehaviour where T : Component
```

Comment:
Please change the name of the class and Give GenericSingleton or MonoGenericSingleton like this . This class name should not be Game Controller one .